python/plugins/processing/algs/saga/SagaAlgorithmProvider.py

```diff
@@ -91,34 +81,28 @@ def _loadAlgorithms(self):
             ProcessingLog.addToLog(ProcessingLog.LOG_ERROR,
                                    self.tr('Problem with SAGA installation: SAGA was not found or is not correctly installed'))
             return
-        if version not in self.supportedVersions:
-            lastVersion = sorted(self.supportedVersions.keys())[-1]
-            if version > lastVersion:
-                version = lastVersion
-            else:
-                ProcessingLog.addToLog(ProcessingLog.LOG_ERROR,
-                                       self.tr('Problem with SAGA installation: installed SAGA version (%s) is not supported' % version))
-                return
+
+        if version not in ['2.3.0', '2.3.1']:
```

Comment:
confirmed what reported by @saberraz 
fresh compilation of SAGA 2.3.1, working fine as standalone. 
Processing *read* SAGA and lists all the algorithms, but when trying to run any of the algorith, this is the error:

```
Traceback (most recent call last):
  File "/home/matteo/lavori/QGIS/build-qgis3/output/python/plugins/processing/gui/ProcessingToolbox.py", line 257, in executeAlgorithm
    alg = alg.getCopy()
  File "/home/matteo/lavori/QGIS/build-qgis3/output/python/plugins/processing/algs/saga/SagaAlgorithm.py", line 78, in getCopy
    newone = SagaAlgorithm212(self.descriptionFile)
NameError: name 'SagaAlgorithm212' is not defined
```